core/src/serve_repair.rs

```diff
@@ -940,66 +925,6 @@ mod tests {
         Blockstore::destroy(&ledger_path).expect("Expected successful database destruction");
     }
 
-    #[test]
-    fn run_orphan_corrupted_shred_size() {
-        solana_logger::setup();
-        let recycler = PacketsRecycler::default();
-        let ledger_path = get_tmp_ledger_path!();
-        {
-            let blockstore = Arc::new(Blockstore::open(&ledger_path).unwrap());
-            // Create slots [1, 2] with 1 shred apiece
-            let (mut shreds, _) = make_many_slot_entries(1, 2, 1);
-
-            // Make shred for slot 1 too large
-            assert_eq!(shreds[0].slot(), 1);
-            assert_eq!(shreds[0].index(), 0);
-            shreds[0].payload.push(10);
-            blockstore
-                .insert_shreds(shreds, None, false)
-                .expect("Expect successful ledger write");
-            let nonce = 42;
-            // Make sure repair response is corrupted
-            assert!(repair_response::repair_response_packet(
-                &blockstore,
-                1,
-                0,
-                &socketaddr_any!(),
-                nonce,
-            )
-            .is_none());
-
-            // Orphan request for slot 2 should only return slot 1 since
-            // calling `repair_response_packet` on slot 1's shred will
-            // be corrupted
-            let rv: Vec<_> = ServeRepair::run_orphan(
-                &recycler,
-                &socketaddr_any!(),
-                Some(&blockstore),
-                2,
-                5,
-                nonce,
-            )
-            .expect("run_orphan packets")
-            .packets
-            .iter()
-            .cloned()
-            .collect();
-
-            // Verify responses
-            let expected = vec![repair_response::repair_response_packet(
-                &blockstore,
-                2,
-                0,
-                &socketaddr_any!(),
-                nonce,
-            )
-            .unwrap()];
-            assert_eq!(rv, expected);
-        }
-
-        Blockstore::destroy(&ledger_path).expect("Expected successful database destruction");
-    }
-
```

Comment:
@carllin - This test previously "corrupted" a shred by add an extra byte which made it unable to fit in a packet. `repair_response_pakcet(...)` would return `None` only when the `repair_response_packet(...)` eventually makes it to this snippet:
```
pub fn repair_response_packet_from_shred(
    shred: Vec<u8>,
    dest: &SocketAddr,
    nonce: Nonce,
) -> Option<Packet> {
    let mut packet = Packet::default();
    packet.meta.size = shred.len() + SIZE_OF_NONCE;
    if packet.meta.size > packet.data.len() {
        return None;
    }
```

With the addition of the size check on insertion into the blockstore, this is no longer possible (and so the test was failing). With that in mind, this tests appears to have been redundant to the test above it, but let me know what you think

Reply:
hmmm by "size check", I think you mean the newly added truncation of the data shred here: https://github.com/solana-labs/solana/pull/16602/files#diff-01b72272fd1033dcbb8fed13821a6232d35913119af6be20a7c2d1e9c772feb9R1422 ?

With these changes, the shred is successfully inserted because its truncated down to `shred.data_header.size`, regardless of the actual data size, and when we fetch from the blockstore later we get a "correctly" sized shred. 

I think if we 
1) Did this https://github.com/solana-labs/solana/pull/16602/files#r617935210,
2) Expand the `shred.data_header.size` of the corrupted shred in this test to trigger that above check

Then this test would succeed because the corrupted shred would never be inserted into Blockstore. That way we can keep this test that ensures corrupted shreds don't make it into repair. Wdyt?

Reply:
> hmmm by "size check", I think you mean the newly added truncation of the data shred here: https://github.com/solana-labs/solana/pull/16602/files#diff-01b72272fd1033dcbb8fed13821a6232d35913119af6be20a7c2d1e9c772feb9R1422 ?

Yes, apologies for the loose wording on my side

> I think if we
> 1. Did this https://github.com/solana-labs/solana/pull/16602/files#r617935210,
> 2. Expand the shred.data_header.size of the corrupted shred in this test to trigger that above check
>
> Then this test would succeed because the corrupted shred would never be inserted into Blockstore. That way we can keep this test that ensures corrupted shreds don't make it into repair. Wdyt?

I had it in my mind that we "had to" insert the shred in this instance. But, truncating a corrupted shred to make it "fit" would do more harm than good. I see what you meant now (I think) with your original comment. See:
https://github.com/solana-labs/solana/pull/16602/commits/35b4da4f5b39990dc3fa3d429161315c633eaca2#diff-01b72272fd1033dcbb8fed13821a6232d35913119af6be20a7c2d1e9c772feb9R1418-R1425